src/socket.cpp

```diff
@@ -40,6 +40,9 @@ void CSocket::Init ( const quint16 iPortNumber )
 
     // create the UDP socket
     UdpSocket = socket ( AF_INET, SOCK_DGRAM, 0 );
+    //
+    const char tos = (const char) iQosNumber;  // Quality of Service
+    setsockopt ( UdpSocket, IPPROTO_IP, IP_TOS, &tos, sizeof(tos) );
```

Comment:
This will prioritise all traffic on all sockets, even the purely informational traffic between server directory (central server) and server.  For an approach that at least tries to confine the prioritisation to sockets where urgent traffic happens (though not distinguishing on a per-message basis which would, for efficiency reasons, require using a raw socket where changing prioritisation on a per-message basis does not come at the cost of a system call), see https://github.com/dakhubgit/jamulus/commit/d206159e60caf6d21d3b1821248a7f9a1d9cff15

Reply:
This is currently how Jamulus does things... in bulk as UDP. Note that audio is > 99.9% of the traffic. Chat messages and server lists may, some day, be separated. Note that iQosNumber can be intercepted before it gets here. Thus, this code already is heading the way you want... but that is not for me.

Reply:
Did you look at the commit I specified?  The whole point was that the IP_TOS for a CSocket was only switched for those sockets that had at least one CHighPrioSocket open on them.  Why would a directory server be contacted via a CHighPrioSocket?

Reply:
@dakhubgit : Why would a directory server be contacted via a CHighPrioSocket?
Thats the way Jamulus was before I wandered past.

Reply:
@dakhubgit All the Any Genre directory servers are all processing audio priority network traffic from multiple clients - they're normal servers providing a real-time jam platform, _as well_.  All traffic uses the same sockets currently, only differentiated at the application protocol layer.

(The genre-specific directory servers only have a single client connection allowed and I run them at lower CPU priority than my jam server.)